### PR TITLE
add wait to create snapshot function

### PR DESCRIPTION
**Pull request by meldafrawi**
Signed-off-by: Mohamed Eldafrawi <mohamed.eldafrawi@rancher.com>

### CODE REVIEW

File: manager/integration/tests/common.py [hunk 1]
```diff
@@ -2433,3 +2433,24 @@ def wait_for_engine_image_deletion(client, core_api, engine_image_name):
...
     assert deleted
+
+
+def create_snapshot(longhorn_api_client, volume_name):
+    volume = longhorn_api_client.by_id_volume(volume_name)
+
+    snapshots = volume.snapshotList(volume=volume_name)
+    snapshots_count = len(snapshots.data)
+
+    volume.snapshotCreate()
```
Comment (yasker) [inline manager/integration/tests/common.py]: snapshotCreate() should return the name of the snapshot as a part of response, which can be used to determine if the snapshot has been created or not.

File: manager/integration/tests/common.py [hunk 2]
```diff
@@ -2433,3 +2433,24 @@ def wait_for_engine_image_deletion(client, core_api, engine_image_name):
...
+    for i in range(RETRY_COUNTS):
+        snapshots = volume.snapshotList(volume=volume_name)
+
+        if len(snapshots) == snapshots_count + 1:
+            snapshot_created = True
+            break
+        time.sleep(RETRY_INTERVAL)
+
+    assert snapshot_created
+    return snapshots
```
Comment (yasker) [inline manager/integration/tests/common.py]: Doesn't make sense for the `create_snapshot` to return a list of snapshots. Make more sense to only return the newly created snapshot's name.

**Comment by yasker** [comment]
We should also replace the existing `volume.snapshotCreate()` call with this call.

File: manager/integration/tests/common.py (hunk 1 above)
Comment (meldafrawi) [reply manager/integration/tests/common.py]: fixed

File: manager/integration/tests/common.py (hunk 2 above)
Comment (meldafrawi) [reply manager/integration/tests/common.py]: fixed

**Comment by yasker** [comment]
LGTM. Need to fix the conflict.

**Comment by meldafrawi** [comment]
conflict resolved.

**Pull request merged**